libs/platform/src/lib/approval-flow/approval-flow-user-list/approval-flow-user-list.component.ts

```diff
@@ -90,4 +97,32 @@ export class ApprovalFlowUserListComponent implements AfterViewInit {
             this.users.find(user => `${this._idPrefix + user.id}` === item.itemEl.nativeElement.id)
         );
     }
+
+    private _collectDataProgressive(): void {
+        const ITEMS_RENDERED_AT_ONCE = 100;
+        const INTERVAL_IN_MS = 10;
+
+        this._displayUsers = [];
+
+        let currentIndex = 0;
+        const interval = setInterval(() => {
+            const nextIndex = currentIndex + ITEMS_RENDERED_AT_ONCE;
+
+            const collectedUsers = [];
+
+            for (let i = currentIndex; i <= nextIndex; i++) {
+                if (i >= this.users.length) {
+                    clearInterval(interval);
+                    break;
+                }
+
+                collectedUsers.push(this.users[i]);
+            }
+
+            currentIndex += ITEMS_RENDERED_AT_ONCE;
+
+            this._displayUsers.push(...collectedUsers);
+            this._cdr.detectChanges();
```

Comment:
is it a reason to not use `.markForCheck()`? detectChanges will trigger re-rendering each 10ms according to your interval